src/gui/accountsettings.cpp

```diff
@@ -797,7 +794,9 @@ void AccountSettings::slotAccountStateChanged()
                 errors << tr("The server version %1 is unsupported! Proceed at your own risk.").arg(account->capabilities().status().versionString());
             }
             showConnectionLabel(tr("Connected to %1.").arg(serverWithUser), errors);
-            ui->openBrowserButton->setVisible(false);
+            if (_askForOAuthLoginDialog != nullptr) {
+                _askForOAuthLoginDialog->accept();
```

Comment:
Why accept?

Reply:
There's no particular reason. We can also just `hide()` or `close()` it if you like to. In my opinion, this makes sense to some extent, as logging in in the browser is similar to clicking a submit button in a basic auth form.

Reply:
Accept and close will trigger the corresponding signals to be emitted.
Anyway you should add a comment that we are hiding the dialog here.
Maybe deleteLater would be correct I'm not exactly sure.

Reply:
if hide deletes the dialog :D